api/src/main/java/com/community/api/endpoint/avisoft/controller/cart/CartEndPoint.java

```diff
@@ -126,21 +127,34 @@ public ResponseEntity<?> retrieveCartItemsCount(@RequestParam long customerId, @
             return new ResponseEntity<>("Error retrieving cart", HttpStatus.INTERNAL_SERVER_ERROR);
         }
     }
+    @Transactional
+    @JsonBackReference
     @RequestMapping(value = "preview-cart", method = RequestMethod.GET)
     public ResponseEntity<?> retrieveCartItems(@RequestParam long customerId, @RequestParam Long orderId) {
         try {
             if (isAnyServiceNull()) {
                 return new ResponseEntity<>("One or more Serivces not initialized",HttpStatus.INTERNAL_SERVER_ERROR);
             }
             Customer customer = customerService.readCustomerById(customerId);
-            if (customer != null) {
-                Order cart=orderService.findOrderById(orderId);
-                if ( cart!= null) {
-                    return new ResponseEntity<>(cart.getOrderItems().get(0).getName(), HttpStatus.OK);
-                } else
-                    return new ResponseEntity<>("No items found", HttpStatus.NOT_FOUND);
+            if (customer == null)
+            {
+                return new ResponseEntity<>("customer does not exist",HttpStatus.NOT_FOUND);
+            }
+            Order cart=orderService.findOrderById(orderId);
```

Comment:
As i can see same queries has been used for multiple times
so create component and add these functions and call from their so that can it be used in module

    public Customer getCustomer(Long customerId) {

        Customer customer = customerService.readCustomerById(customerId);

        if (customer == null) {

            throw new CustomerNotFoundException("Customer not found for this Id");

        }

        return customer;

    }



    public Order getCart(Customer customer) {

        Order cart = orderService.findCartForCustomer(customer);

        if (cart == null) {

            cart = orderService.createNewCartForCustomer(customer);

        }

        return cart;

    }



    public Product getProduct(Long productId) {

        Product product = catalogService.findProductById(productId);

        if (product == null) {

            throw new ProductNotFoundException("Product not found");

        }

        return product;

    }